src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -6,30 +6,45 @@
 
 import com.revrobotics.CANSparkMax;
 import com.revrobotics.SparkAbsoluteEncoder;
+import com.revrobotics.CANSparkBase.IdleMode;
 import com.revrobotics.CANSparkLowLevel.MotorType;
+
+import edu.wpi.first.wpilibj.DigitalInput;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.HopperConstants;
 import frc.robot.Ports;
+import frc.robot.Ports.HopperPorts;
 
 public class Intake extends SubsystemBase {
-  /** Creates a new Intake. */
+  //intake
   // private static CANSparkMax rotationNEO;
   private static CANSparkMax rollerNEO;
   private static SparkAbsoluteEncoder encoder;
   // private static PIDController intakePIDController;
-  // private static boolean isRunning;
-  // private double setpoint;
+  private static boolean isRunning;
+  //private double setpoint;
+
+  //hopper
+  private final DigitalInput beamBreakIntake;
+  private final DigitalInput beamBreakShooter;
+  private final CANSparkMax hopperMotor;
 
   public Intake() {
+    //intake
     // rotationNEO = new CANSparkMax(Ports.IntakePorts.rotationNEOPort,
     // MotorType.kBrushless);
     rollerNEO = new CANSparkMax(Ports.IntakePorts.rollerNEOPort, MotorType.kBrushless);
-    // encoder = rotationNEO.getAbsoluteEncoder(Type.kDutyCycle);
-    // intakePIDController = new
-    // PIDController(Constants.IntakeConstants.PIDConstants.kP,
-    // Constants.IntakeConstants.PIDConstants.kI,
-    // Constants.IntakeConstants.PIDConstants.kD);
-    // setpoint = encoder.getPosition();
-    // isRunning = false;
+    //encoder = rotationNEO.getAbsoluteEncoder(Type.kDutyCycle); 
```

Comment:
lines 37-39; pid no longer necessary; intake is stationary

Reply:
still need pid controller for velocity pid; instead of absolute, use the built-in relative encoders with .getEncoder()